src/include/distributed/multi_logical_replication.h

```diff
@@ -12,58 +12,155 @@
 #ifndef MULTI_LOGICAL_REPLICATION_H_
 #define MULTI_LOGICAL_REPLICATION_H_
 
+#include "c.h"
 
 #include "nodes/pg_list.h"
+#include "distributed/connection_management.h"
 
 
 /* Config variables managed via guc.c */
 extern int LogicalReplicationTimeout;
 
 extern bool PlacementMovedUsingLogicalReplicationInTX;
 
+/*
+ * NodeAndOwner should be used as a key for structs that should be hashed by a
+ * combination of node and owner.
+ */
+typedef struct NodeAndOwner
+{
+	uint32_t nodeId;
+	Oid tableOwnerId;
+} NodeAndOwner;
+
+
+/*
+ * ReplicationSlotInfo stores the info that defines a replication slot. For
+ * shard splits this information is built by parsing the result of the
+ * 'worker_split_shard_replication_setup' UDF.
+ */
+typedef struct ReplicationSlotInfo
+{
+	uint32 targetNodeId;
+	Oid tableOwnerId;
+	char *name;
+} ReplicationSlotInfo;
+
+/*
+ * PublicationInfo stores the information that defines a publication.
+ */
+typedef struct PublicationInfo
+{
+	NodeAndOwner key;
+	char *name;
+	List *shardIntervals;
+	struct SubscriptionInfo *subscription;
+} PublicationInfo;
+
+/*
+ * Stores information necesary for creating a subscription
+ */
+typedef struct SubscriptionInfo
+{
+	char *name;
+	Oid tableOwnerId;
+
+	/*
+	 * The name of the user that's used as the owner of the subscription.
+	 */
+	char *temporaryOwnerName;
+	ReplicationSlotInfo *replicationSlot;
+	PublicationInfo *publication;
+
+	/*
+	 * The shardIntervals that this subscription is meant to create. For shard
+	 * splits this can be different than the shards that are part of the
+	 * publication, because of the existence of dummy shards.
+	 */
+	List *newShards;
```

Comment:
Is there any special reason to add newShards as a part of SubscriptionInfo? Usually, the subsription would be binded to a publication and under the publication the corresponding tables would be abstracted.

Reply:
I changed the name of SubscriptionInfo to LogicalRepTarget to make this less confusing.